### PR TITLE
Version 0.17.32b

### PR DESCRIPTION
**Pull request by BossCode45**
What I changed:
 - Buffed slide boost: It now reduces the dash cooldown (can be disabled in slide boost animation). This was done to encourage the use of more slide boosts in speedruns.
- Perhaps fixed item persistence error: I changed nothing but when I built and ran it it worked.
- Updated project unity version to the latest version of unity

Known bugs:
 - Player hit sound doesn't play.

### CODE REVIEW

File: Assets/BoostSlide.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    override public void OnStateEnter(Animator animator, AnimatorStateInfo stateInfo, int layerIndex)
+    {
+        pc = animator.GetComponent<PlayerController>();
+        if (pc.dashCooldown && shorterDashCooldown)
+        {
+            pc.reduceDashCooldown();
+        }
+    }
+
+    // OnStateUpdate is called on each Update frame between OnStateEnter and OnStateExit callbacks
```
Comment (garzaa) [inline Assets/BoostSlide.cs:19]: You can remove these comments.

File: Assets/Scripts/Player/AttackBuffer.cs [hunk 2]
```diff
@@ -88,5 +88,5 @@ public enum AttackDirection {
 public enum AttackType {
     NONE = 0,
     PUNCH = 1,
-    KICK = 2
+    KICK = 2,
```
Comment (garzaa) [inline Assets/Scripts/Player/AttackBuffer.cs:91]: Nice

File: Assets/Scripts/Player/PlayerController.cs [hunk 3]
```diff
@@ -10,7 +10,7 @@ public class PlayerController : Entity {
 	const float hardLandDistance = .64f * 4f;
 	const float dashSpeed = 6f;
 	const float terminalFallSpeed = -10f;
-	const float dashCooldownLength = .6f;
+	public static float dashCooldownLength = .6f;
```
Comment (garzaa) [inline Assets/Scripts/Player/PlayerController.cs:13]: Does this need to be public now? Looks like it's just being referenced in the PlayerController still.

File: Assets/Scripts/Player/PlayerController.cs [hunk 4]
```diff
@@ -646,20 +646,20 @@ public class PlayerController : Entity {
 		anim.SetTrigger("Dash");
 	}
 
-	public void StartDashAnimation(bool backwards) {
+	public void StartDashAnimation(bool backwards, bool recycle = false) {
 		preDashSpeed = Mathf.Abs(rb2d.velocity.x);
 
 		// back dash animation always comes from the initial dash, where the speed boost has already been applied
-		float newSpeed = (backwards ? preDashSpeed : preDashSpeed+dashSpeed);
```
Comment (garzaa) [inline Assets/Scripts/Player/PlayerController.cs:653]: I kept the spacing here tight to not interfere with my mental parsing of that ternary conditional operator.

File: Assets/Scripts/Player/PlayerController.cs [hunk 5]
```diff
@@ -678,6 +678,10 @@ public class PlayerController : Entity {
 		if (dashTimeout != null) StopCoroutine(dashTimeout);
 	    dashTimeout = StartCoroutine(RunDashCooldown(dashCooldownLength));
 	}
+	public void reduceDashCooldown()
```
Comment (garzaa) [inline Assets/Scripts/Player/PlayerController.cs:681]: Class methods in C# should start with an uppercase letter btw

File: Assets/Scripts/Player/PlayerController.cs [hunk 6]
```diff
@@ -708,7 +712,7 @@ public class PlayerController : Entity {
 		anim.SetBool("JustFlipped", false);
 	}
 
-	IEnumerator RunDashCooldown(float seconds) {
+	public IEnumerator RunDashCooldown(float seconds) {
```
Comment (garzaa) [inline Assets/Scripts/Player/PlayerController.cs:715]: I also don't see this referenced anywhere outside of PlayerController, I think it's safe to make it non-public.

File: Assets/BoostSlide.cs [hunk 7]
```diff
@@ -0,0 +1,47 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BoostSlide : StateMachineBehaviour
+{
+    PlayerController pc;
+    public bool shorterDashCooldown;
```
Comment (garzaa) [inline Assets/BoostSlide.cs:8]: This is only going to be changed at editor-time, so it's safe to hard-set it to true or false (unless you're putting this on other state machine states).

**Review by garzaa — CHANGES_REQUESTED** [review]
This PR is pretty large for the intended behavior (dash slide fix and item getting fix), I'd suggest repeating it with just running `git add *.cs *.cs.meta Assets/Player/PlayerAnimator.controller` so it's just a few files. The Unity player update should be a separate commit so it's not tied to bugfixes.

You were right about the item thing also being fixed already. You'll notice an extra pumpkin in Deep Town which is part of a cutscene I was working on, that was partly what's keeping the update delayed.

Anyway, thanks very much for this, and being willing to mentally parse the Vapor Trails codebase. I'll merge your next PR with these suggested changes.

File: Assets/Scripts/Player/PlayerController.cs (hunk 3 above)
Comment (BossCode45) [reply Assets/Scripts/Player/PlayerController.cs:13]: Yeah that was from a previous idea of how to implement the slide buff.

File: Assets/Scripts/Player/PlayerController.cs (hunk 5 above)
Comment (BossCode45) [reply Assets/Scripts/Player/PlayerController.cs:681]: Oh ok.

File: Assets/Scripts/Player/PlayerController.cs (hunk 6 above)
Comment (BossCode45) [reply Assets/Scripts/Player/PlayerController.cs:715]: Yeah that was from a previous idea of how to implement the slide buff.

File: Assets/Scripts/Player/PlayerController.cs (hunk 4 above)
Comment (BossCode45) [reply Assets/Scripts/Player/PlayerController.cs:653]: Visual studio doesn't like your thinking lol.

File: Assets/Scripts/Player/AttackBuffer.cs (hunk 2 above)
Comment (BossCode45) [reply Assets/Scripts/Player/AttackBuffer.cs:91]: Another idea that didn't really work out.

File: Assets/BoostSlide.cs (hunk 7 above)
Comment (BossCode45) [reply Assets/BoostSlide.cs:8]: Just did that for visibility.

**Pull request closed without merging**